conn/controlfns_windows.go

```diff
@@ -0,0 +1,21 @@
+/* SPDX-License-Identifier: MIT
+ *
+ * Copyright (C) 2017-2022 WireGuard LLC. All Rights Reserved.
+ */
+
+package conn
+
+import "syscall"
+
+func init() {
+	controlFns = append(controlFns,
+		// Set SO_RCVBUF/SO_SNDBUF - this could be common with the _unix code except
+		// for the unfortunate type specificity of syscall.Handle.
+		func(network, address string, c syscall.RawConn) error {
+			return c.Control(func(fd uintptr) {
```

Comment:
Does this actually do anything on Windows, given that Windows uses WinRIO? It might, I just don't know myself.

Reply:
Yes, at least for API users who use StdNetBind, such as Tailscale.

Reply:
`SO_{RCV,SND}BUF` seems like a new thing. Can you separate its use out into a different commit with its own justification?

(It can stay in this PR, but just have a separate commit with a message I can read to understand this.)

Reply:
The socket options for UDP snd/rcv buffer size have been broken out into their own commit, with these details included in the commit message https://github.com/WireGuard/wireguard-go/commit/aeed2bac0755cc595ffb6f7d248f4c8c2a61b8e7.